### PR TITLE
Fix generate dropdown menu

### PR DESCRIPTION
**Pull request by art009**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: src/Nav.php [hunk 1]
```diff
@@ -252,7 +252,7 @@ private function renderItem(array $item): string
         if (!isset($item['label'])) {
             throw new InvalidArgumentException('The "label" option is required.');
         }
-
+        $this->dropdown = false;
```
Comment (Mister-42) [inline src/Nav.php:255]: Set [by default](https://github.com/yiisoft/yii-bulma/blob/master/src/Nav.php#L21)

File: src/Nav.php [hunk 2]
```diff
@@ -304,9 +304,10 @@ private function renderItem(array $item): string
 
         if ($this->dropdown) {
             return
-                html::beginTag('div', $options) . "\n" .
+                Html::beginTag('div', $options) . "\n" .
                     Html::a($label, $url, ['class' => 'navbar-link']) . "\n" .
-                    $items;
+                    $items. "\n" .
+                Html::endTag('div');
```
Comment (Mister-42) [inline src/Nav.php:310]: It's in a bit of a weird place, but that's already done [here](https://github.com/yiisoft/yii-bulma/blob/master/src/Nav.php#L30)

**Review by Mister-42 — CHANGES_REQUESTED** [review]

File: src/Nav.php (hunk 1 above)
Comment (art009) [reply src/Nav.php:255]: This variable neads reset for each item, because it sets submenu in only this item.

File: src/Nav.php (hunk 2 above)
Comment (art009) [reply src/Nav.php:310]: This tag closes submenu in item.

File: src/Nav.php (hunk 1 above)
Comment (art009) [reply src/Nav.php:255]: Next item after dropdown, was wraped in div. ( [code](https://github.com/yiisoft/yii-bulma/blob/master/src/Nav.php#L305) )

**Comment by Mister-42** [comment]
Ok, let's see what your code outputs (example taken from NavTest::testNavIcon)
```php
        $html = NavBar::widget()
            ->brandLabel('My Project')
            ->brandImage('yii-logo.jpg')
            ->brandUrl('/')
            ->options(['class' => 'is-black', 'data-sticky' => '', 'data-sticky-shadow' => ''])
            ->itemsOptions(['class' => 'navbar-end'])
            ->begin();

        $html .= Nav::widget()
            ->items([
                [
                    'label' => 'Setting Account',
                    'url' => '/setting/account',
                    'icon' => 'fas fa-user-cog',
                    'iconOptions' => ['class' => 'icon'],
                ],
                [
                    'label' => 'Profile',
                    'url' => '/profile',
                    'icon' => 'fas fa-users',
                    'iconOptions' => ['class' => 'icon'],
                ],
                [
                    'label' => 'Admin' . Html::img(
                        '../../docs/images/icon-avatar.png',
                        ['class' => 'img-rounded', 'aria-expanded' => 'false']
                    ),
                    'items' => [
                        ['label' => 'Logout', 'url' => '/auth/logout'],
                    ],
                    'encode' => false,
                ],
            ])
            ->render();

        $html .= NavBar::end();
        echo $html;
```

```html
<nav id="w1-navbar" class="navbar is-black" data-sticky="" data-sticky-shadow="">
<div class="navbar-brand"><span class="navbar-item"><img src="yii-logo.jpg" alt=""></span><a class="navbar-item" href="/">My Project</a><a class="navbar-burger" aria-expanded="false" aria-label="menu" role="button"><span aria-hidden="true"></span><span aria-hidden="true"></span><span aria-hidden="true"></span></a></div>
<div id="w1-navbar-Menu" class="navbar-menu"><div class="navbar-end"><a class="navbar-item" href="/setting/account"><span class="icon"><i class="fas fa-user-cog"></i></span><span>Setting Account</span></a>
<a class="navbar-item" href="/profile"><span class="icon"><i class="fas fa-users"></i></span><span>Profile</span></a>
<div class="navbar-item has-dropdown is-hoverable">
<a class="navbar-link" href="#">Admin<img class="img-rounded" src="../../docs/images/icon-avatar.png" alt="" aria-expanded="false"></a>
<div id="w2-dropdown" class="navbar-dropdown">
<a class="navbar-item" href="/auth/logout">Logout</a>
</div>

</div></div></div>
</div>
</nav>
```

That is wrong and has a </div> too many.

**Comment by Mister-42** [comment]
Or the smaller example from NavTest::testNavDropdownLinkDisabled

```php
        echo Nav::widget()
            ->encodeLabels(true)
            ->items([
                [
                    'label' => 'Link disable',
                    'url' => '#',
                    'items' => [
                        ['label' => 'Level 1', 'url' => '#', 'disabled' => true],
                        ['label' => 'Level 2', 'url' => '#'],
                    ],
                ],
            ])
            ->render();
```

```html
<div class="navbar-item has-dropdown is-hoverable">
<a class="navbar-link" href="#">Link disable</a>
<div id="w1-dropdown" class="navbar-dropdown">
<a class="navbar-item" href="#" style="opacity:.65; pointer-events:none;">Level 1</a>
<a class="navbar-item" href="#">Level 2</a>
</div>

</div></div>
```

File: src/Nav.php (hunk 2 above)
Comment (art009) [reply src/Nav.php:305]: > 
> 
> It's in a bit of a weird place, but that's already done [here](https://github.com/yiisoft/yii-bulma/blob/master/src/Nav.php#L30)

Really, a weird place)) remove

**Comment by art009** [comment]
> 
> 
> Or the smaller example from NavTest::testNavDropdownLinkDisabled
> 
Fix, now code
```
<a class="navbar-item" href="/setting/account"><span class="icon"><i class="fas fa-user-cog"></i></span><span>Setting Account</span></a>
<div class="navbar-item has-dropdown is-hoverable">
	<a class="navbar-link" href="#">Admin<img class="img-rounded" src="../../docs/images/icon-avatar.png" alt="" aria-expanded="false"></a>
	<div id="w2-dropdown" class="navbar-dropdown">
		<a class="navbar-item" href="/auth/logout">Logout</a>
	</div>
</div>
<a class="navbar-item" href="/profile"><span class="icon"><i class="fas fa-users"></i></span><span>Profile</span></a>
```

**Comment by Mister-42** [comment]
Yes, that is fixed.
According to the static tests, the only different this PR is making is adding 1 additional newline.

File: src/Nav.php [hunk 3]
```diff
@@ -24,13 +24,7 @@ final class Nav extends Widget
...
     protected function run(): string
     {
-        $html = $this->renderItems();
-
-        if ($this->dropdown) {
-            $html .= Html::endTag('div');
-        }
-
-        return $html;
+        return $this->renderItems();
```
Comment (samdark) [inline src/Nav.php:27]: There's no need for separate `renderItems()` method, the code could be moved to `run()`. The change itself is alright.

File: src/Nav.php [hunk 4]
```diff
@@ -252,7 +246,7 @@ private function renderItem(array $item): string
         if (!isset($item['label'])) {
             throw new InvalidArgumentException('The "label" option is required.');
         }
-
+        $this->dropdown = false;
```
Comment (samdark) [inline src/Nav.php:249]: @art009 can we pass it to the method instead of using class property?

File: src/Nav.php (hunk 3 above)
Comment (art009) [reply src/Nav.php]: Move.

File: src/Nav.php (hunk 4 above)
Comment (art009) [reply src/Nav.php]: Delete property.

**Comment by samdark** [comment]
Thank you.

**Pull request merged**